### PR TITLE
Set/Delete Props - New tests, bug fix & improvements

### PR DESCRIPTION
**Pull request by RobiNino**

### CODE REVIEW

File: src/main/java/org/jfrog/hudson/pipeline/declarative/steps/generic/EditPropsStep.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        void editPropsRun(Run build, TaskListener listener, EditPropsStep step, FilePath ws, EnvVars env) throws IOException, InterruptedException {
+            // Set Artifactory server
+            org.jfrog.hudson.pipeline.common.types.ArtifactoryServer pipelineServer = DeclarativePipelineUtils
+                    .getArtifactoryServer(build, ws, getContext(), step.serverId);
+            artifactoryServer = Utils.prepareArtifactoryServer(null, pipelineServer);
+
+            // Set spec
+            SpecConfiguration specConfiguration = new SpecConfiguration(step.spec, step.specPath);
+            spec = SpecUtils.getSpecStringFromSpecConf(specConfiguration, env, ws, listener.getLogger());
+
```
Comment (yahavi) [inline src/main/java/org/jfrog/hudson/pipeline/declarative/steps/generic/EditPropsStep.java]:
```suggestion
```

File: src/main/java/org/jfrog/hudson/pipeline/declarative/steps/generic/EditPropsStep.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+public class EditPropsStep extends AbstractStepImpl {
+    protected String serverId;
+    protected String spec;
+    private String props;
+    private String specPath;
+    private boolean failNoOp;
+    private EditPropertiesActionType editType;
+
+    @DataBoundConstructor
+    public EditPropsStep(String serverId, EditPropertiesActionType editType) {
```
Comment (yahavi) [inline src/main/java/org/jfrog/hudson/pipeline/declarative/steps/generic/EditPropsStep.java]:
```suggestion
    EditPropsStep(String serverId, EditPropertiesActionType editType) {
```

File: src/test/resources/integration/pipelines/declarative/setProps.pipeline [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    stage "Set Props"
+    // Set property on one of the files (a.in)
+    rtSetProps(
+            serverId: serverId,
+            spec: """{
+                       "files": [
+                        {
+                            "pattern": "${LOCAL_REPO1}/a.in"
+                          }
```
Comment (yahavi) [inline src/test/resources/integration/pipelines/declarative/setProps.pipeline]:
```suggestion
                        }
```

File: src/main/java/org/jfrog/hudson/pipeline/declarative/steps/generic/GenericStep.java [hunk 4]
```diff
@@ -34,8 +34,7 @@
     private String specPath;
     boolean failNoOp;
 
-    @DataBoundConstructor
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/pipeline/declarative/steps/generic/GenericStep.java]: Was this annotation removed intentionally?

Comment (RobiNino) [reply src/main/java/org/jfrog/hudson/pipeline/declarative/steps/generic/GenericStep.java]: Yes, these two steps are never called directly

**Pull request merged**